src/Controller/RegistrationController.php

```diff
@@ -57,4 +57,45 @@ public function registerAction(
         }
         return $this->render('registration/first_user.html.twig', ['form' => $form->createView()]);
     }
+
+    /**
+     * @Route("/register", name="app_register")
+     * @param Request $request
+     * @param UserPasswordEncoderInterface $passwordEncoder
+     * @param GuardAuthenticatorHandler $guardHandler
+     * @param LoginFormAuthenticator $authenticator
+     * @return Response
+     */
+    public function register(
+        Request $request,
+        UserPasswordEncoderInterface $passwordEncoder,
+        GuardAuthenticatorHandler $guardHandler,
+        LoginFormAuthenticator $authenticator
+    ): Response {
+
+        $user = new User();
+        $form = $this->createForm(RegistrationFormType::class, $user);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            // encode the plain password
+            $user->setPassword(
+                $passwordEncoder->encodePassword(
+                    $user,
+                    $form->get('password')->getData()
+                )
+            );
+            $user->setRoles(["ROLE_USER"]);
+            $entityManager = $this->getDoctrine()->getManager();
```

Comment:
You should use dependency injection to get services (like the EntityManager) in controllers